### PR TITLE
Managed VPC resource

### PR DESCRIPTION
**Pull request by tbroden84**
Handle standalone VPC as managed VPC resource in the provider. Part of decouple the VPCs from the instances and will be released as v1.16.0. 

Note:
- Introducing managed VPC resource to decouple VPC from instance. 
- To avoid breaking changes 
    - It's still possible to create VPC from instance with vpc_subnet, but is discouraged.
    - Default behaviour for instance is still to delete associated VPC. Set attribute keep_associated_vpc to true, in order to override the default behaviour when deleting an instance.

Features:
- data_source/account_vpcs, list all available standalone VPCs for an account
- data_source/vpc_gcp_info, either use vpc_id or instance_id to fetch information
- data_source/vpc_info, either use vpc_id or instance_id to fetch information
- resource/instance, keep_associated_vpc set to true to keep managed VPC when deleting instance.
- resource/vpc, managed VPC resource
- resource/vpc_gcp_peering, either use vpc_id or instance_id as dependency for this resource
- resource/vpc_peering, either use vpc_id or instance_id as dependency for this resource

Deprecated:
- data_source/vpc_gcp_info, intance_id use vpc_id instead
- data_source/vpc_info, instance_id use vpc_id instead
- resource/instance, vpc_subnet create managed VPC instead
- resource/vpc_gcp_peering, intance_id use vpc_id instead
- resource/vpc_peering, intance_id use vpc_id instead

Documentation:
- guides/managed_vpc, added new documentation with additional examples for managed VPC for v1.16.0
- guides/plan, prioritized new subscriptions plans over legacy
- data_source/account, updated attribute references
- data_source/account_vpcs, added new documentation
- data_source/vpc_gcp_info, added v1.16.0 information and collapsable examples
- data_source/vpc_info, added v1.16.0 information and collapsable examples
- resource/instance, added v1.16.0 attribute information and collapsable examples
- resource/vpc, added new documentation
- resource/vpc_gcp_peering, added v1.16.0 information and collapsable examples
- resource/vpc_peering, added v1.16.0 information and collapsable examples

Make sure to deploy go-api v1.6.0 before and run `go mod tidy` as release preparation.

Issue #133

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ local-clean:  ## Clean files
 	rm -rf ~/.terraform.d/plugins/localhost/cloudamqp/cloudamqp/$(PROVIDER_VERSION)/darwin_amd64/*
 
 ## Local clean, build and install
-local-build: local-clean
+local-build: #local-clean
```
Comment (dentarg) [inline GNUmakefile:24]: Should this be committed?

Comment (tbroden84) [reply GNUmakefile:24]: Perhaps not.

File: cloudamqp/resource_cloudamqp_vpc.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+				Description: "VPC name given when hosted at the cloud provider",
+			},
+			// "instances": {
+			// 	Type:     schema.TypeList,
+			// 	Computed: true,
+			// 	Elem: &schema.Schema{
+			// 		Type: schema.TypeInt,
+			// 	},
+			// 	Description: "Collection of instance identifiers connected to the VPC",
+			// },
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_vpc.go:58]: Yes

File: cloudamqp/resource_cloudamqp_vpc.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	}
+
+	data, err := api.CreateVpcInstance(params)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] cloudamqp::vpc::create data: %v", data)
+	d.SetId(data["id"].(string))
+	//d.Set("vpc_name", data["name"].(string))
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_vpc.go:80]: Yes

File: cloudamqp/resource_cloudamqp_vpc.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+func resourceVpcDelete(d *schema.ResourceData, meta interface{}) error {
+	api := meta.(*api.API)
+
+	// Todo: Remove this check and trust the response from API?
+	// Cannot update this resource with instances changes before running an extra
+	// "terraform plan" or "terraform apply --refresh-only"
+	// instances := d.Get("instances").([]interface{})
+	// if len(instances) > 0 {
+	// 	return fmt.Errorf("VPC still contains instances: %v", instances)
+	// }
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_vpc.go:130]: Yes

File: cloudamqp/resource_cloudamqp_vpc.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+func validateVpcSchemaAttribute(key string) bool {
+	switch key {
+	case "name",
+		"region",
+		"subnet",
+		"tags",
+		"vpc_name":
+		// "instances":
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_vpc.go:141]: yes

File: cloudamqp/resource_cloudamqp_vpc_gcp_peering.go [hunk 6]
```diff
@@ -53,7 +59,19 @@ func resourceCreateVpcGcpPeering(d *schema.ResourceData, meta interface{}) error
...
+	// data, err := api.RequestVpcGcpPeering(d.Get("instance_id").(int), params)
+	log.Printf("[DEBUG] cloudamqp::vpc_gcp_peering::create instance_id: %v, vpc_id: %v", d.Get("instance_id"), d.Get("vpc_id"))
+	data := make(map[string]interface{})
+	err := errors.New("")
+	if d.Get("instance_id") == 0 && d.Get("vpc_id") == nil {
+		return errors.New("Missing both instance identifier or vpc identifier")
+	} else if d.Get("instance_id") != 0 {
+		data, err = api.RequestVpcGcpPeering(d.Get("instance_id").(int), params)
+	} else if d.Get("vpc_id") != nil {
+		data, err = api.RequestVpcGcpPeeringTemp(d.Get("vpc_id").(string), params)
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_vpc_gcp_peering.go:72]: Fixed in latest go-api v1.6.0 release

File: go.mod [hunk 7]
```diff
@@ -89,4 +88,4 @@ require (
 	google.golang.org/protobuf v1.25.0 // indirect
 )
 
-//replace github.com/84codes/go-api => ../../84codes/go-api
+replace github.com/84codes/go-api => ../../84codes/go-api
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply go.mod:91]: My bad

File: cloudamqp/resource_cloudamqp_vpc_gcp_peering.go [hunk 8]
```diff
@@ -53,7 +59,19 @@ func resourceCreateVpcGcpPeering(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
-	data, err := api.RequestVpcGcpPeering(d.Get("instance_id").(int), params)
+	// Todo: Create if/else if/else to check either instance_id or vpc_id. Different calls!
+	// data, err := api.RequestVpcGcpPeering(d.Get("instance_id").(int), params)
```
[the remark this answers is not in the document]
Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_vpc_gcp_peering.go:63]: yes

**Pull request merged**